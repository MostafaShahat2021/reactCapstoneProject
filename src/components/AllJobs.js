import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaAngleRight } from 'react-icons/fa';

function AllJobs() {
  const jobs = useSelector((state) => state.jobs.jobs);
  // console.log(jobs);
  const [search, setSearch] = useState('');

  const handleSearch = (e) => {
    const searchValue = e.target.value;
    setSearch(searchValue);
  };

  const filterJobs = jobs.filter((job) => job.title.toLowerCase().includes(search.toLowerCase()));
  return (
    <>
      <h1>Remotive jobs</h1>
      <h3>
        Remote Jobs Avilabele
        <span>
          {jobs.length}
        </span>
      </h3>
      <input
        className="search"
        type="text"
        role="searchbox"
        placeholder="Search jobs"
        value={search}
        onChange={(e) => handleSearch(e)}
      />
      <div className="cards">
        {filterJobs.map((job) => (
          <div className="card" key={job.id}>
            <img
              src={job?.company_logo}
              alt={job.title}
              className="card-image"
            />
            <div className="card-content">
              <h2 className="card-title">{job.title}</h2>
              <p className="card-description">{job.category}</p>
              <Link
                onClick={() => {
                  const jobId = job.id;
                  window.location.href = `/details/${jobId}`;
                }}
                to="/details"
              >
                {/* View Details */}
                <FaAngleRight className="FaAngleRight" />
              </Link>
            </div>
          </div>
        ))}
      </div>
    </>
  );
}

export default AllJobs;
