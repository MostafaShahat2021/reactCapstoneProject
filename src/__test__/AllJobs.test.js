import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AllJobs from '../components/AllJobs';

const mockStore = configureStore([]);

describe('AllJobs component', () => {
  it('render correctly', () => {
    const store = mockStore({
      jobs: {
        jobs: [],
      },
    });

    const tree = renderer.create(
      <Provider store={store}>
        <AllJobs />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
