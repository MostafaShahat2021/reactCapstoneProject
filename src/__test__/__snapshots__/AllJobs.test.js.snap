// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllJobs component render correctly 1`] = `
Array [
  <main
    className="hero"
  >
    <h1>
      Remotive jobs
    </h1>
    <img
      alt="logo"
      className="card-image"
      src="https://camo.githubusercontent.com/5a759c5f7640123e386607aea7890caf84fb7ce9f2a0888407e0d218cc86bf82/68747470733a2f2f72656d6f746976652e636f6d2f6c6f676f"
    />
  </main>,
  <h3>
    Remote Jobs Avilabele
    <span>
      0
    </span>
  </h3>,
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search jobs"
    role="searchbox"
    type="text"
    value=""
  />,
  <section
    className="cards"
  />,
]
`;
